File: java-vertx/src/main/java/io/swagger/server/api/model/V3ResultOutlet.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3ResultOutlet
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3ResultOutlet   {
  private Float outletDistance = null;

  private String outletSlidSpid = null;

  private String outletName = null;

  private String outletBusiness = null;

  private Float outletLatitude = null;

  private Float outletLongitude = null;

  private String outletSuburb = null;

  private Integer outletPostcode = null;

  private String outletBusinessHourMon = null;

  private String outletBusinessHourTue = null;

  private String outletBusinessHourWed = null;

  private String outletBusinessHourThur = null;

  private String outletBusinessHourFri = null;

  private String outletBusinessHourSat = null;

  private String outletBusinessHourSun = null;

  private String outletNotes = null;

  public V3ResultOutlet outletDistance(Float outletDistance) {
    this.outletDistance = outletDistance;
    return this;
  }

  /**
   * Distance of outlet from input location (in metres); returns 0 if no location is input
   * @return outletDistance
   **/
    public Float getOutletDistance() {
    return outletDistance;
  }

  public void setOutletDistance(Float outletDistance) {
    this.outletDistance = outletDistance;
  }

  public V3ResultOutlet outletSlidSpid(String outletSlidSpid) {
    this.outletSlidSpid = outletSlidSpid;
    return this;
  }

  /**
   * The SLID / SPID
   * @return outletSlidSpid
   **/
    public String getOutletSlidSpid() {
    return outletSlidSpid;
  }

  public void setOutletSlidSpid(String outletSlidSpid) {
    this.outletSlidSpid = outletSlidSpid;
  }

  public V3ResultOutlet outletName(String outletName) {
    this.outletName = outletName;
    return this;
  }

  /**
   * The location name of the outlet
   * @return outletName
   **/
    public String getOutletName() {
    return outletName;
  }

  public void setOutletName(String outletName) {
    this.outletName = outletName;
  }

  public V3ResultOutlet outletBusiness(String outletBusiness) {
    this.outletBusiness = outletBusiness;
    return this;
  }

  /**
   * The business name of the outlet
   * @return outletBusiness
   **/
    public String getOutletBusiness() {
    return outletBusiness;
  }

  public void setOutletBusiness(String outletBusiness) {
    this.outletBusiness = outletBusiness;
  }

  public V3ResultOutlet outletLatitude(Float outletLatitude) {
    this.outletLatitude = outletLatitude;
    return this;
  }

  /**
   * Geographic coordinate of latitude at outlet
   * @return outletLatitude
   **/
    public Float getOutletLatitude() {
    return outletLatitude;
  }

  public void setOutletLatitude(Float outletLatitude) {
    this.outletLatitude = outletLatitude;
  }

  public V3ResultOutlet outletLongitude(Float outletLongitude) {
    this.outletLongitude = outletLongitude;
    return this;
  }

  /**
   * Geographic coordinate of longitude at outlet
   * @return outletLongitude
   **/
    public Float getOutletLongitude() {
    return outletLongitude;
  }

  public void setOutletLongitude(Float outletLongitude) {
    this.outletLongitude = outletLongitude;
  }

  public V3ResultOutlet outletSuburb(String outletSuburb) {
    this.outletSuburb = outletSuburb;
    return this;
  }

  /**
   * The city/municipality the outlet is in
   * @return outletSuburb
   **/
    public String getOutletSuburb() {
    return outletSuburb;
  }

  public void setOutletSuburb(String outletSuburb) {
    this.outletSuburb = outletSuburb;
  }

  public V3ResultOutlet outletPostcode(Integer outletPostcode) {
    this.outletPostcode = outletPostcode;
    return this;
  }

  /**
   * The postcode for the outlet
   * @return outletPostcode
   **/
    public Integer getOutletPostcode() {
    return outletPostcode;
  }

  public void setOutletPostcode(Integer outletPostcode) {
    this.outletPostcode = outletPostcode;
  }

  public V3ResultOutlet outletBusinessHourMon(String outletBusinessHourMon) {
    this.outletBusinessHourMon = outletBusinessHourMon;
    return this;
  }

  /**
   * The business hours on Monday
   * @return outletBusinessHourMon
   **/
    public String getOutletBusinessHourMon() {
    return outletBusinessHourMon;
  }

  public void setOutletBusinessHourMon(String outletBusinessHourMon) {
    this.outletBusinessHourMon = outletBusinessHourMon;
  }

  public V3ResultOutlet outletBusinessHourTue(String outletBusinessHourTue) {
    this.outletBusinessHourTue = outletBusinessHourTue;
    return this;
  }

  /**
   * The business hours on Tuesday
   * @return outletBusinessHourTue
   **/
    public String getOutletBusinessHourTue() {
    return outletBusinessHourTue;
  }

  public void setOutletBusinessHourTue(String outletBusinessHourTue) {
    this.outletBusinessHourTue = outletBusinessHourTue;
  }

  public V3ResultOutlet outletBusinessHourWed(String outletBusinessHourWed) {
    this.outletBusinessHourWed = outletBusinessHourWed;
    return this;
  }

  /**
   * The business hours on Wednesday
   * @return outletBusinessHourWed
   **/
    public String getOutletBusinessHourWed() {
    return outletBusinessHourWed;
  }

  public void setOutletBusinessHourWed(String outletBusinessHourWed) {
    this.outletBusinessHourWed = outletBusinessHourWed;
  }

  public V3ResultOutlet outletBusinessHourThur(String outletBusinessHourThur) {
    this.outletBusinessHourThur = outletBusinessHourThur;
    return this;
  }

  /**
   * The business hours on Thursday
   * @return outletBusinessHourThur
   **/
    public String getOutletBusinessHourThur() {
    return outletBusinessHourThur;
  }

  public void setOutletBusinessHourThur(String outletBusinessHourThur) {
    this.outletBusinessHourThur = outletBusinessHourThur;
  }

  public V3ResultOutlet outletBusinessHourFri(String outletBusinessHourFri) {
    this.outletBusinessHourFri = outletBusinessHourFri;
    return this;
  }

  /**
   * The business hours on Friday
   * @return outletBusinessHourFri
   **/
    public String getOutletBusinessHourFri() {
    return outletBusinessHourFri;
  }

  public void setOutletBusinessHourFri(String outletBusinessHourFri) {
    this.outletBusinessHourFri = outletBusinessHourFri;
  }

  public V3ResultOutlet outletBusinessHourSat(String outletBusinessHourSat) {
    this.outletBusinessHourSat = outletBusinessHourSat;
    return this;
  }

  /**
   * The business hours on Saturday
   * @return outletBusinessHourSat
   **/
    public String getOutletBusinessHourSat() {
    return outletBusinessHourSat;
  }

  public void setOutletBusinessHourSat(String outletBusinessHourSat) {
    this.outletBusinessHourSat = outletBusinessHourSat;
  }

  public V3ResultOutlet outletBusinessHourSun(String outletBusinessHourSun) {
    this.outletBusinessHourSun = outletBusinessHourSun;
    return this;
  }

  /**
   * The business hours on Sunday
   * @return outletBusinessHourSun
   **/
    public String getOutletBusinessHourSun() {
    return outletBusinessHourSun;
  }

  public void setOutletBusinessHourSun(String outletBusinessHourSun) {
    this.outletBusinessHourSun = outletBusinessHourSun;
  }

  public V3ResultOutlet outletNotes(String outletNotes) {
    this.outletNotes = outletNotes;
    return this;
  }

  /**
   * Any additional notes for the outlet such as 'Buy pre-loaded myki cards only'. May be null/empty.
   * @return outletNotes
   **/
    public String getOutletNotes() {
    return outletNotes;
  }

  public void setOutletNotes(String outletNotes) {
    this.outletNotes = outletNotes;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3ResultOutlet v3ResultOutlet = (V3ResultOutlet) o;
    return Objects.equals(this.outletDistance, v3ResultOutlet.outletDistance) &&
        Objects.equals(this.outletSlidSpid, v3ResultOutlet.outletSlidSpid) &&
        Objects.equals(this.outletName, v3ResultOutlet.outletName) &&
        Objects.equals(this.outletBusiness, v3ResultOutlet.outletBusiness) &&
        Objects.equals(this.outletLatitude, v3ResultOutlet.outletLatitude) &&
        Objects.equals(this.outletLongitude, v3ResultOutlet.outletLongitude) &&
        Objects.equals(this.outletSuburb, v3ResultOutlet.outletSuburb) &&
        Objects.equals(this.outletPostcode, v3ResultOutlet.outletPostcode) &&
        Objects.equals(this.outletBusinessHourMon, v3ResultOutlet.outletBusinessHourMon) &&
        Objects.equals(this.outletBusinessHourTue, v3ResultOutlet.outletBusinessHourTue) &&
        Objects.equals(this.outletBusinessHourWed, v3ResultOutlet.outletBusinessHourWed) &&
        Objects.equals(this.outletBusinessHourThur, v3ResultOutlet.outletBusinessHourThur) &&
        Objects.equals(this.outletBusinessHourFri, v3ResultOutlet.outletBusinessHourFri) &&
        Objects.equals(this.outletBusinessHourSat, v3ResultOutlet.outletBusinessHourSat) &&
        Objects.equals(this.outletBusinessHourSun, v3ResultOutlet.outletBusinessHourSun) &&
        Objects.equals(this.outletNotes, v3ResultOutlet.outletNotes);
  }

  @Override
  public int hashCode() {
    return Objects.hash(outletDistance, outletSlidSpid, outletName, outletBusiness, outletLatitude, outletLongitude, outletSuburb, outletPostcode, outletBusinessHourMon, outletBusinessHourTue, outletBusinessHourWed, outletBusinessHourThur, outletBusinessHourFri, outletBusinessHourSat, outletBusinessHourSun, outletNotes);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3ResultOutlet {\n");
    
    sb.append("    outletDistance: ").append(toIndentedString(outletDistance)).append("\n");
    sb.append("    outletSlidSpid: ").append(toIndentedString(outletSlidSpid)).append("\n");
    sb.append("    outletName: ").append(toIndentedString(outletName)).append("\n");
    sb.append("    outletBusiness: ").append(toIndentedString(outletBusiness)).append("\n");
    sb.append("    outletLatitude: ").append(toIndentedString(outletLatitude)).append("\n");
    sb.append("    outletLongitude: ").append(toIndentedString(outletLongitude)).append("\n");
    sb.append("    outletSuburb: ").append(toIndentedString(outletSuburb)).append("\n");
    sb.append("    outletPostcode: ").append(toIndentedString(outletPostcode)).append("\n");
    sb.append("    outletBusinessHourMon: ").append(toIndentedString(outletBusinessHourMon)).append("\n");
    sb.append("    outletBusinessHourTue: ").append(toIndentedString(outletBusinessHourTue)).append("\n");
    sb.append("    outletBusinessHourWed: ").append(toIndentedString(outletBusinessHourWed)).append("\n");
    sb.append("    outletBusinessHourThur: ").append(toIndentedString(outletBusinessHourThur)).append("\n");
    sb.append("    outletBusinessHourFri: ").append(toIndentedString(outletBusinessHourFri)).append("\n");
    sb.append("    outletBusinessHourSat: ").append(toIndentedString(outletBusinessHourSat)).append("\n");
    sb.append("    outletBusinessHourSun: ").append(toIndentedString(outletBusinessHourSun)).append("\n");
    sb.append("    outletNotes: ").append(toIndentedString(outletNotes)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: swagger.2.0/openapi/csharp/src/Ptv.Api.Client/Model/V3ResultOutlet.cs
/*
 * PTV Timetable API - Version 3
 *
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * The version of the OpenAPI document: v3
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Ptv.Api.Client.Client.OpenAPIDateConverter;

namespace Ptv.Api.Client.Model
{
    /// <summary>
    /// V3ResultOutlet
    /// </summary>
    [DataContract(Name = "V3.ResultOutlet")]
    public partial class V3ResultOutlet : IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="V3ResultOutlet" /> class.
        /// </summary>
        /// <param name="outletDistance">Distance of outlet from input location (in metres); returns 0 if no location is input.</param>
        /// <param name="outletSlidSpid">The SLID / SPID.</param>
        /// <param name="outletName">The location name of the outlet.</param>
        /// <param name="outletBusiness">The business name of the outlet.</param>
        /// <param name="outletLatitude">Geographic coordinate of latitude at outlet.</param>
        /// <param name="outletLongitude">Geographic coordinate of longitude at outlet.</param>
        /// <param name="outletSuburb">The city/municipality the outlet is in.</param>
        /// <param name="outletPostcode">The postcode for the outlet.</param>
        /// <param name="outletBusinessHourMon">The business hours on Monday.</param>
        /// <param name="outletBusinessHourTue">The business hours on Tuesday.</param>
        /// <param name="outletBusinessHourWed">The business hours on Wednesday.</param>
        /// <param name="outletBusinessHourThur">The business hours on Thursday.</param>
        /// <param name="outletBusinessHourFri">The business hours on Friday.</param>
        /// <param name="outletBusinessHourSat">The business hours on Saturday.</param>
        /// <param name="outletBusinessHourSun">The business hours on Sunday.</param>
        /// <param name="outletNotes">Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty..</param>
        public V3ResultOutlet(float outletDistance = default(float), string outletSlidSpid = default(string), string outletName = default(string), string outletBusiness = default(string), float outletLatitude = default(float), float outletLongitude = default(float), string outletSuburb = default(string), int outletPostcode = default(int), string outletBusinessHourMon = default(string), string outletBusinessHourTue = default(string), string outletBusinessHourWed = default(string), string outletBusinessHourThur = default(string), string outletBusinessHourFri = default(string), string outletBusinessHourSat = default(string), string outletBusinessHourSun = default(string), string outletNotes = default(string))
        {
            this.OutletDistance = outletDistance;
            this.OutletSlidSpid = outletSlidSpid;
            this.OutletName = outletName;
            this.OutletBusiness = outletBusiness;
            this.OutletLatitude = outletLatitude;
            this.OutletLongitude = outletLongitude;
            this.OutletSuburb = outletSuburb;
            this.OutletPostcode = outletPostcode;
            this.OutletBusinessHourMon = outletBusinessHourMon;
            this.OutletBusinessHourTue = outletBusinessHourTue;
            this.OutletBusinessHourWed = outletBusinessHourWed;
            this.OutletBusinessHourThur = outletBusinessHourThur;
            this.OutletBusinessHourFri = outletBusinessHourFri;
            this.OutletBusinessHourSat = outletBusinessHourSat;
            this.OutletBusinessHourSun = outletBusinessHourSun;
            this.OutletNotes = outletNotes;
        }

        /// <summary>
        /// Distance of outlet from input location (in metres); returns 0 if no location is input
        /// </summary>
        /// <value>Distance of outlet from input location (in metres); returns 0 if no location is input</value>
        [DataMember(Name = "outlet_distance", EmitDefaultValue = false)]
        public float OutletDistance { get; set; }

        /// <summary>
        /// The SLID / SPID
        /// </summary>
        /// <value>The SLID / SPID</value>
        [DataMember(Name = "outlet_slid_spid", EmitDefaultValue = false)]
        public string OutletSlidSpid { get; set; }

        /// <summary>
        /// The location name of the outlet
        /// </summary>
        /// <value>The location name of the outlet</value>
        [DataMember(Name = "outlet_name", EmitDefaultValue = false)]
        public string OutletName { get; set; }

        /// <summary>
        /// The business name of the outlet
        /// </summary>
        /// <value>The business name of the outlet</value>
        [DataMember(Name = "outlet_business", EmitDefaultValue = false)]
        public string OutletBusiness { get; set; }

        /// <summary>
        /// Geographic coordinate of latitude at outlet
        /// </summary>
        /// <value>Geographic coordinate of latitude at outlet</value>
        [DataMember(Name = "outlet_latitude", EmitDefaultValue = false)]
        public float OutletLatitude { get; set; }

        /// <summary>
        /// Geographic coordinate of longitude at outlet
        /// </summary>
        /// <value>Geographic coordinate of longitude at outlet</value>
        [DataMember(Name = "outlet_longitude", EmitDefaultValue = false)]
        public float OutletLongitude { get; set; }

        /// <summary>
        /// The city/municipality the outlet is in
        /// </summary>
        /// <value>The city/municipality the outlet is in</value>
        [DataMember(Name = "outlet_suburb", EmitDefaultValue = false)]
        public string OutletSuburb { get; set; }

        /// <summary>
        /// The postcode for the outlet
        /// </summary>
        /// <value>The postcode for the outlet</value>
        [DataMember(Name = "outlet_postcode", EmitDefaultValue = false)]
        public int OutletPostcode { get; set; }

        /// <summary>
        /// The business hours on Monday
        /// </summary>
        /// <value>The business hours on Monday</value>
        [DataMember(Name = "outlet_business_hour_mon", EmitDefaultValue = false)]
        public string OutletBusinessHourMon { get; set; }

        /// <summary>
        /// The business hours on Tuesday
        /// </summary>
        /// <value>The business hours on Tuesday</value>
        [DataMember(Name = "outlet_business_hour_tue", EmitDefaultValue = false)]
        public string OutletBusinessHourTue { get; set; }

        /// <summary>
        /// The business hours on Wednesday
        /// </summary>
        /// <value>The business hours on Wednesday</value>
        [DataMember(Name = "outlet_business_hour_wed", EmitDefaultValue = false)]
        public string OutletBusinessHourWed { get; set; }

        /// <summary>
        /// The business hours on Thursday
        /// </summary>
        /// <value>The business hours on Thursday</value>
        [DataMember(Name = "outlet_business_hour_thur", EmitDefaultValue = false)]
        public string OutletBusinessHourThur { get; set; }

        /// <summary>
        /// The business hours on Friday
        /// </summary>
        /// <value>The business hours on Friday</value>
        [DataMember(Name = "outlet_business_hour_fri", EmitDefaultValue = false)]
        public string OutletBusinessHourFri { get; set; }

        /// <summary>
        /// The business hours on Saturday
        /// </summary>
        /// <value>The business hours on Saturday</value>
        [DataMember(Name = "outlet_business_hour_sat", EmitDefaultValue = false)]
        public string OutletBusinessHourSat { get; set; }

        /// <summary>
        /// The business hours on Sunday
        /// </summary>
        /// <value>The business hours on Sunday</value>
        [DataMember(Name = "outlet_business_hour_sun", EmitDefaultValue = false)]
        public string OutletBusinessHourSun { get; set; }

        /// <summary>
        /// Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty.
        /// </summary>
        /// <value>Any additional notes for the outlet such as &#39;Buy pre-loaded myki cards only&#39;. May be null/empty.</value>
        [DataMember(Name = "outlet_notes", EmitDefaultValue = false)]
        public string OutletNotes { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class V3ResultOutlet {\n");
            sb.Append("  OutletDistance: ").Append(OutletDistance).Append("\n");
            sb.Append("  OutletSlidSpid: ").Append(OutletSlidSpid).Append("\n");
            sb.Append("  OutletName: ").Append(OutletName).Append("\n");
            sb.Append("  OutletBusiness: ").Append(OutletBusiness).Append("\n");
            sb.Append("  OutletLatitude: ").Append(OutletLatitude).Append("\n");
            sb.Append("  OutletLongitude: ").Append(OutletLongitude).Append("\n");
            sb.Append("  OutletSuburb: ").Append(OutletSuburb).Append("\n");
            sb.Append("  OutletPostcode: ").Append(OutletPostcode).Append("\n");
            sb.Append("  OutletBusinessHourMon: ").Append(OutletBusinessHourMon).Append("\n");
            sb.Append("  OutletBusinessHourTue: ").Append(OutletBusinessHourTue).Append("\n");
            sb.Append("  OutletBusinessHourWed: ").Append(OutletBusinessHourWed).Append("\n");
            sb.Append("  OutletBusinessHourThur: ").Append(OutletBusinessHourThur).Append("\n");
            sb.Append("  OutletBusinessHourFri: ").Append(OutletBusinessHourFri).Append("\n");
            sb.Append("  OutletBusinessHourSat: ").Append(OutletBusinessHourSat).Append("\n");
            sb.Append("  OutletBusinessHourSun: ").Append(OutletBusinessHourSun).Append("\n");
            sb.Append("  OutletNotes: ").Append(OutletNotes).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
